include/ck/tensor_operation/gpu/element/trinary_element_wise_operation.hpp

```diff
@@ -0,0 +1,283 @@
+    {
+        const float a = x0 + x1 + x2;
+        y             = a > type_convert<half_t>(0.0f) ? a : type_convert<half_t>(0.0f);
+    };
+
+    template <>
+    __host__ __device__ constexpr void operator()<float, float, half_t, float>(
+        float& y, const float& x0, const half_t& x1, const float& x2) const
+    {
+        const float a = x0 + type_convert<float>(x1) + x2;
+        y             = a > 0.0f ? a : 0.0f;
+    };
+};
+
+struct TriAddHardswish
+{
+    template <typename T>
+    __host__ __device__ constexpr void
+    operator()(T& y, const T& x0, const T& x1, const T& x2) const;
+
+    template <>
+    __host__ __device__ constexpr void
+    operator()<float>(float& y, const float& x0, const float& x1, const float& x2) const
+    {
+        float a = x0 + x1 + x2;
+        float b = a + float{3};
+        float c = (b > 0) * (b > 6.0f ? 6.0f : b) * a * 0.166667f;
+        y       = c;
+    };
+
+    template <>
+    __host__ __device__ constexpr void
+    operator()<double>(double& y, const double& x0, const double& x1, const double& x2) const
+    {
+        double a = x0 + x1 + x2;
+        double b = a + 3.0;
+        double c = (b > 0) * (b > 6.0 ? 6.0 : b) * a * 0.166667;
+        y        = c;
+    };
+
+    template <>
+    __host__ __device__ constexpr void
+    operator()<half_t>(half_t& y, const half_t& x0, const half_t& x1, const half_t& x2) const
+    {
+        float a = x0 + x1 + x2;
+        float b = a + 3.0f;
+        float c = (b > 0) * (b > 6.0f ? 6.0f : b) * a * 0.166667f;
+        y       = c;
+    };
+};
+
+// C = A * B
+// E = FastGelu(C + D)
+// TODO: Change for trinary elementwise
+struct TriAddFastGelu
+{
+    // Fast GeLU
+    // https://paperswithcode.com/method/gelu
+    // y = 0.5*x*(1+tanh(sqrt(2/pi)*(x+0.044715*x^3)))
+    __host__ __device__ static constexpr float GetFastGeLU(float x)
+    {
+        const float u   = 2.f * x * (0.035677f * x * x + 0.797885f);
+        const float emu = exp(-u);
+        const float cdf = 0.5f + 0.5f * (2.f / (1.f + emu) - 1.f);
+        return x * cdf;
+    }
+
+    template <typename T>
+    static inline constexpr bool is_valid_param_type_v =
+        std::is_same_v<T, float> || std::is_same_v<T, half_t> || std::is_same_v<T, bhalf_t> ||
+        std::is_same_v<T, int32_t> || std::is_same_v<T, int8_t>;
+
+    template <typename E, typename C, typename D>
+    __host__ __device__ constexpr void operator()(E& e, const C& c, const D& d) const
+    {
+        static_assert(is_valid_param_type_v<E> && is_valid_param_type_v<C> &&
+                      is_valid_param_type_v<D>);
+
+        const float y = GetFastGeLU(type_convert<float>(c) + type_convert<float>(d));
+
+        e = type_convert<E>(y);
+    }
+
+    template <typename D>
+    __host__ __device__ constexpr void operator()(float& e, const float& c, const D& d) const
+    {
+        static_assert(is_valid_param_type_v<D>);
+
+        e = GetFastGeLU(c + type_convert<float>(d));
+    }
```

Comment:
I'm sorry for intervention, I've just starting to look into CK repository and decided to check a few PRs.
My question is why we need the second `operator()` specialized with `floats` while the first one perfectly covers that case?
`type_convert` under the hood does `static_cast` and if `E` and `C` types are `float` then it just does nothing like the code in the second `operator()`. Or am I missed something?